src/algorithms/digi/LGADPulseDigitization.h

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Souvik Paul, Chun Yuen Tsang, Prithwish Tribedy
+// Special Acknowledgement: Kolja Kauder
+//
+// Convert ADC pulses from LGADPulseGeneration into ADC and TDC values
+
+#pragma once
+
+#include <algorithms/algorithm.h>
+#include <edm4eic/RawTrackerHitCollection.h>
+#include <edm4hep/RawTimeSeriesCollection.h>
+#include <string>
+#include <string_view>
+
+#include "algorithms/digi/LGADHitDigiConfig.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+using LGADPulseDigitizationAlgorithm =
+    algorithms::Algorithm<algorithms::Input<edm4hep::RawTimeSeriesCollection>,
+                          algorithms::Output<edm4eic::RawTrackerHitCollection>>;
+
+class LGADPulseDigitization : public LGADPulseDigitizationAlgorithm,
+                             public WithPodConfig<LGADHitDigiConfig> {
```

Comment:
Either `LGADPulseDigitization` and `LGADPulseDigitizationConfig` or `LGADHitDigi` and `LGADHitDigiConfig`.

Reply:
I have separated the Config classes for `LGADPulseDigitization `and `LGADPulseGeneration`. From a physics standpoint, some of the parameters like `tdc_range `and `adc_range `should be shared between the two classes, so ideally some parameters should be shared. But I will leave them separate here.

Reply:
Let's stick to convention that each algorithm has a separate config. It's okay to have multiple config structures have identical fields.

Reply:
Perfect! The newest commit has already satisfied this requirement.